File: Java Project  V1.7/backend-spring/src/main/java/com/emart/entity/ProductImage.java
package com.emart.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

/**
 * Entity representing a single product image.
 * A product can have multiple images (one-to-many).
 *
 * Table: product_image_master
 */
@Entity
@Table(name = "product_image_master")
public class ProductImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "image_id")
    private Long imageId;


    // =========================
    // PRODUCT (FK)
    // =========================

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prod_id", nullable = false)
    @NotNull(message = "Product is required for an image")
    private Product product;


    // =========================
    // IMAGE URL
    // =========================

    @NotBlank(message = "Image URL is required")
    @Column(name = "image_url", nullable = false, length = 500)
    private String imageUrl;


    // =========================
    // DEFAULT CONSTRUCTOR
    // =========================

    public ProductImage() {
    }


    // =========================
    // GETTERS AND SETTERS
    // =========================

    public Long getImageId() {
        return imageId;
    }

    public void setImageId(Long imageId) {
        this.imageId = imageId;
    }


    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }


    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: DotNet Project V2.0/backend-dotnet/src/Emart.Domain/Entities/ProductImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Emart.Domain.Entities;

/// <summary>
/// A single product image. A product can have multiple images (one-to-many).
/// Table: product_image_master.
/// </summary>
[Table("product_image_master")]
public class ProductImage
{
    [Key]
    [Column("image_id")]
    public long ImageId { get; set; }

    [Required]
    [Column("prod_id")]
    public long ProductId { get; set; }

    public Product Product { get; set; } = null!;

    [Required]
    [MaxLength(500)]
    [Column("image_url")]
    public string ImageUrl { get; set; } = null!;
}
